src/main/java/uk/gov/indexer/dao/CurrentKeysUpdateDAO.java

```diff
@@ -10,10 +10,38 @@
 
 @UseStringTemplate3StatementLocator
 interface CurrentKeysUpdateDAO extends DBConnectionDAO {
-    String CURRENT_KEYS_TABLE = "CURRENT_KEYS";
 
-    @SqlUpdate("CREATE TABLE IF NOT EXISTS " + CURRENT_KEYS_TABLE + " (KEY VARCHAR PRIMARY KEY, SERIAL_NUMBER INTEGER UNIQUE)")
-    void ensureCurrentKeysTableExists();
+    String CURRENT_KEYS_TABLE = "current_keys";
+    String TOTAL_RECORDS_TABLE = "total_records";
+    String TOTAL_RECORDS_FUNCTION = TOTAL_RECORDS_TABLE + "_fn()";
+    String TOTAL_RECORDS_TRIGGER = TOTAL_RECORDS_TABLE + "_trigger";
+
+    @SqlUpdate(
+            "CREATE TABLE IF NOT EXISTS " + CURRENT_KEYS_TABLE + " (KEY VARCHAR PRIMARY KEY, SERIAL_NUMBER INTEGER UNIQUE);" +
+
+                    "CREATE TABLE IF NOT EXISTS " + TOTAL_RECORDS_TABLE + " (COUNT INTEGER);" +
+
+                    //Insert query below initializes the total records for pre existing register by setting the value as no of rows in current_keys table
+                    "INSERT INTO " + TOTAL_RECORDS_TABLE + "(COUNT) SELECT (SELECT COUNT(*) FROM " + CURRENT_KEYS_TABLE + ") WHERE NOT EXISTS(SELECT 1 FROM " + TOTAL_RECORDS_TABLE + ");" +
+
+                    "CREATE OR REPLACE FUNCTION " + TOTAL_RECORDS_FUNCTION + " RETURNS TRIGGER\n" +
+                    "AS $$\n" +
+                    "BEGIN\n" +
+                    "  IF TG_OP = 'INSERT' THEN\n" +
+                    "     EXECUTE 'UPDATE " + TOTAL_RECORDS_TABLE + " SET COUNT=COUNT + 1';\n" +
+                    "     RETURN NEW;\n" +
+                    "  END IF;\n" +
+                    "  RETURN NULL;\n" +
+                    "  END;\n" +
+                    "$$ LANGUAGE plpgsql;" +
+
+                    "DROP TRIGGER IF EXISTS " + TOTAL_RECORDS_TRIGGER + " ON " + CURRENT_KEYS_TABLE + ";" +
+
+                    "CREATE TRIGGER " + TOTAL_RECORDS_TRIGGER + " \n" +
+                    " AFTER INSERT ON " + CURRENT_KEYS_TABLE +
+                    " FOR EACH ROW EXECUTE PROCEDURE " + TOTAL_RECORDS_FUNCTION + ";"
```

Comment:
Yes I thought about it, specially after (https://github.com/openregister/indexer/pull/26).